src/python/pants/java/nailgun_client.py

```diff
@@ -356,22 +364,8 @@ def execute(self, main_class, cwd=None, *args, **environment):
       remote_pgrp_callback=self._receive_remote_pgrp)
     try:
       return self._session.execute(cwd, main_class, *args, **environment)
-    except socket.error as e:
-      raise self.NailgunError(
-        address=self._address_string,
-        pid=self._maybe_last_pid(),
-        pgrp=self._maybe_last_pgrp(),
-        wrapped_exc=e,
-        traceback=sys.exc_info()[2]
-      )
-    except NailgunProtocol.ProtocolError as e:
-      raise self.NailgunError(
-        address=self._address_string,
-        pid=self._maybe_last_pid(),
-        pgrp=self._maybe_last_pgrp(),
-        wrapped_exc=e,
-        traceback=sys.exc_info()[2]
-      )
+    except (socket.error, NailgunProtocol.ProtocolError) as e:
+      raise self._make_nailgun_error(e)
```

Comment:
It looks like the method body could be inlined again to preserve the traceback?

Reply:
Is this really how the traceback is preserved? And they say "explicit is better than implicit" and then don't make you put the exception whose stack you're tracing in the exception constructor? Nonsense.

Reply:
Thanks! Doing that now.

Reply:
(that seems like a particular nonsensical pythonism)